dependencies.qmd

```diff
@@ -34,6 +34,21 @@ We carefully evaluate the potential benefits of data.table as a dependency based
 
 We embrace an [iterative development approach](principles.html#lean-and-agile-collaboration-framework) where dependencies can evolve over time. Initially, dependencies can be included to quickly build a Minimum Viable Product (MVP) and gain early feedback. However, we recognize that not all dependencies may bring significant value in the long-run. Therefore, we encourage package maintainers to regularly evaluate the usefulness and impact of dependencies and remove those that do not contribute substantially to the package. The [itdepends package](https://github.com/r-lib/itdepends) can help to identify dependencies that are used just one time in a given package or codebase. Such dependencies are good candidates for removal.
 
+### Base R support schedule
+
+We maintain a minimum support for the last four versions of R ([the {tidyverse} standard](https://www.tidyverse.org/blog/2019/04/r-version-support/)). The table outlines the concrete minimum supported version.
+
+| Year | Current R version | Minimum supported version |
+|------|-------------------|---------------------------|
+| 2019 | 3.6               | 3.2                       |
+| 2020 | 4.0               | 3.3                       |
+| 2021 | 4.1               | 3.4                       |
+| 2022 | 4.2               | 3.5                       |
```

Comment:
Is it perhaps better to remove these rows, to avoid any ambiguity? We are past them, after all. Perhaps 2023 could also be removed? Also, would there be a way to automate the generation of this table?

Reply:
```suggestion
```

Like this?

Regarding automation - everything can be automated but I wouldn't prioritize it 👇 

![](https://imgs.xkcd.com/comics/automation.png)

Reply:
Yep; unless there's a specific reason to keep these rows?